src/pybind/mgr/alert/module.py

```diff
@@ -0,0 +1,194 @@
+        auth.set_access_token(slef.config['twitter_access_token'], self.config['twitter_access_token_secret'])
+        return tweepy.API(auth)
+
+    def tweets(alert_message):
+        api = get_api()
+        tweet = alert_message
+        status = api.update_status(status=tweet)
+        return status
+
+    def slack(alert_message):
+        slack = Slacker(self.config['slack_api'])
+        post = alert_message
+        return post
+
+    def email(alert_message):
+        msg = MIMEText(alert_message)
+        msg['Subject'] = 'Ceph: Cluster Health Status Alert'
+        msg['From'] = self.config['email_sender']
+        msg['To'] = self.config['email_receiver']
+        server = smtplib.SMTP('{}:{}'.format(self.config['mail_server'], self.config['mail_port']))
+        server.ehlo()
+        server.starttls()
+        server.login(self.config['email_sender'], self.config['sender_password'])
+        sendmail = server.sendmail(self.config['email_sender'], self.config['email_receiver'],msg.as_string())
+        return sendmail
+
+    def new_alert(self, cmd):
+        self.refresh_config()
+        if option in ['slack_api']:    
+
+    def handle_command(self, inbuf, cmd):
+        self.log.error("handle_command")
+        if cmd['prefix'] == 'email alert':
+            return self.alert_via_email(cmd)
+        elif cmd['prefix'] == 'slack alert':
+            return self.post_to_slack(cmd)
+        else:
+            raise NotImplementedError(cmd['prefix'])
+
+    def refresh_config(self):
+        for opt in self.OPTIONS:
+            setattr(self,
+                    opt['name'],
+                    self.get_config(opt['name']) or opt['default'])
+            self.log.debug(' %s = %s', opt['name'], getattr(self, opt['name']))
+
+    def serve(self):
+        """
+        This method is called by the mgr when the module starts and can be
+        used for any background activity.
+        """
+        self.log.info("Starting")
+        old_health = None
+        old_health = json.loads(self.get('health')['json'])
+
+        while self.run:
+            self.refresh_config()
+            if time.time() >= time_last_checked:
+                time_last_checked = time.time()+60
+                new_health = json.loads(self.get('health')['json'])
+                for code, item in new_health['checks'].iteritems():
+                    if code not in old_health['checks']:
+                        # it's a new alert
+                        new[code] = item
+                    else:
+                        old_item = old_health['checks'][code]
+                        if item['severity'] != old_item['severity'] or item['summary']['message'] != old_item['summary']['message']:
+                            # changed
+                            updated[code] = item
+                for code, item in old_health['checks'].iteritems():
+                    if code not in new_health['checks']:
+                        # health alert has resolved
```

Comment:
then after this, something that formats a text alert out of the whole thing.  can start with something trivial just to test, like
```
message = 'new %s\nold %s\nupdated %s\n' % (new, old, updated)
```
and then log.debug(message) for debugging and/or actually send it as an email.

Once this is in place there are a couple directions to go... making the message formatted nicely, with a subject line, and adding support for sending it to different places.

Maybe put this above section in a helper function like generate_message() that you call with somethign like
```
(subject, message) = self.generate_alert(old_health, new_health)
```
?

Reply:
you mean the entire stuff inside the serve function?